### PR TITLE
Version is optional

### LINKED ISSUE #124: Version is optional
**What is the problem you're trying to solve**
As @elrac [pointed out](https://github.com/compose-spec/compose-spec/issues/13#issuecomment-751972799), the spec still defined `version:` as required even though it optional now (per https://github.com/compose-spec/compose-spec/issues/13).

**Describe the solution you'd like**
We should amend the spec to list `version:` as an optional attribute.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by EricHripko**
**What this PR does / why we need it**:
This PR removes any unneeded/leftover mentions of version in the spec and ensures that `version:` attribute is defined as optional.

**Which issue(s) this PR fixes**:

Fixes #124.

### CODE REVIEW

**Comment by EricHripko** [comment]
also CC @kohidave

File: spec.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ will use a platform-specific lookup mechanism to retrieve runtime values.
 ## Compose file
 
 The Compose file is a [YAML](http://yaml.org/) file defining
-[version](#version) (REQUIRED),
+[version](#version) (OPTIONAL),
```
Comment (ndeloof) [inline spec.md:151]: We could even mark this as DEPRECATED

**Review by ndeloof — APPROVED** [review]

File: spec.md (hunk 1 above)
Comment (EricHripko) [reply spec.md]: Great suggestion! Done 👍

**Comment by EricHripko** [comment]
Thanks for landing this @ndeloof 🙌

**Pull request merged**